api/routes.go

```diff
@@ -56,13 +59,23 @@ func InstallRoutes(r *mux.Router, sdkRouter *sdkrouter.Router) {
 	internalRouter.Handle("/metrics", promhttp.Handler())
 
 	v2Router := r.PathPrefix("/api/v2").Subrouter()
-	v2Router.Use(defaultMiddlewares(sdkRouter, config.GetInternalAPIHost()))
+	v2Router.Use(defaultMiddlewares(sdkRouter, authProvider))
 	v2Router.HandleFunc("/status", status.GetStatusV2).Methods(http.MethodGet)
 	v2Router.HandleFunc("/status", emptyHandler).Methods(http.MethodOptions)
+
+	publishHandler, err := publish.NewHandler(authProvider, uploadPath)
+	if err != nil {
+		logger.Log().WithError(err).Fatal(err)
+	}
+
+	publishRouter := v2Router.PathPrefix("/publish").Subrouter()
+	publishRouter.HandleFunc("/", publishHandler.PostFile).Methods(http.MethodPost)
```

Comment:
It seems to me that we can post files completely unathenticated, it should not be the case.

Reply:
it doesn't actually, its a sub router of `v2` that already apply the default middlewares